### PR TITLE
Added eslint rules and installed plugins. Fixed errors where needed

### PR DESCRIPTION
**Pull request by AriHrannar**
Added eslint rules I currently use at work (they are very negotiable though and need to be honed over time)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,43 +1,231 @@
...
+    "plugin:import/errors",
+    "plugin:import/warnings"
   ],
+  "ecmaFeatures": {
+    "modules": true,
+    "jsx": true
+  },
   "parserOptions": {
-    "ecmaVersion": 2016,
+    "ecmaVersion": 8,
```
Comment (Ragnar-Hardarson) [inline .eslintrc]: why the version bump?

File: .eslintrc [hunk 2]
```diff
@@ -1,43 +1,231 @@
...
-    "jsx": true
+      "jsx": true
     }
   },
   "env": {
+    "browser": true,
     "es6": true,
-    "node": true
+    "node": true,
+    "mocha": true
```
Comment (Ragnar-Hardarson) [inline .eslintrc]: why is mocha needed?

File: .eslintrc (hunk 1 above)
Comment (Ragnar-Hardarson) [inline .eslintrc]: And why isn't it consistent with the `env` block below?

File: .eslintrc [hunk 3]
```diff
@@ -1,43 +1,231 @@
...
   },
   "rules": {
-    // Flow
```
Comment (Ragnar-Hardarson) [inline .eslintrc]: Why you hating on flow? 😢

File: .eslintrc [hunk 4]
```diff
@@ -1,43 +1,231 @@
...
-    "flowtype/define-flow-type": 1,
-    "flowtype/no-weak-types": 2,
-    "flowtype/require-valid-file-annotation": 2,
-    "flowtype/type-id-match": [
-    2,
-    "^([A-Z][a-z0-9]+)+Type|Props$"
-    ],
-    "flowtype/use-flow-type": 1,
-    "flowtype/valid-syntax": 1,
-    "prettier/prettier": ["error", {"trailingComma": "es5", "singleQuote": true, "parser": "flow", "printWidth": 90} ]
```
Comment (Ragnar-Hardarson) [inline .eslintrc]: noooo I want prettier

File: .eslintrc (hunk 2 above)
Comment (AriHrannar) [inline .eslintrc]: Its needed for the testing files. "describe" and "it" are mocha functions. If mocha isnt set to true in the env, eslint thinks those 2 functions are undefined and complains :)

Comment (Ragnar-Hardarson) [inline .eslintrc]: We're actually using jest, not mocha.
While they share the "describe" and "it", if we start using more complex jest, this is going to be an issue

**Comment by Ragnar-Hardarson** [comment]
Looks good 😄 

Couple of things:
1. Could you provide a description for the PR? Gives me context now, and in the future if we need to look at this later.
2. I ran https://github.com/prettier/eslint-config-prettier#cli-helper-tool and there are a number of rules already covered by Prettier. Output below

```
➜  veritus-web git:(eslint-configuration) ✗ yarn eslint-check
yarn eslint-check v0.22.0
$ eslint --print-config .eslintrc | eslint-config-prettier-check
The following rules are unnecessary or might conflict with prettier:

- array-bracket-spacing
- arrow-parens
- arrow-spacing
- block-spacing
- brace-style
- comma-dangle
- comma-spacing
- comma-style
- computed-property-spacing
- eol-last
- func-call-spacing
- indent
- jsx-quotes
- key-spacing
- keyword-spacing
- new-parens
- newline-per-chained-call
- no-extra-semi
- no-mixed-spaces-and-tabs
- no-multi-spaces
- no-multiple-empty-lines
- no-trailing-spaces
- no-whitespace-before-property
- one-var-declaration-per-line
- quote-props
- react/jsx-first-prop-new-line
- semi-spacing
- space-before-blocks
- space-in-parens
- space-infix-ops

The following rules can be enabled in some cases. See:
https://github.com/prettier/eslint-config-prettier#special-rules

- curly
- no-confusing-arrow
- no-mixed-operators
error Command failed with exit code 2.
```

**Comment by AriHrannar** [comment]
Ive added a description. Might be a good idea to create github templates to make it more standardized?

Ill look at those rules!

**Comment by AriHrannar** [comment]
Ive removed the rules that conflicted with prettier @Ragnar-Hardarson

File: .eslintrc [hunk 5]
```diff
@@ -10,34 +12,192 @@
...
+    "no-floating-decimal": 2,
+    "no-global-assign": 2,
+    "no-implicit-coercion": 1,
+    "no-implicit-globals": 1,
+    "no-implied-eval": 1,
+    "no-iterator": 2,
+    "no-labels": 1,
+    "no-lone-blocks": 1,
+    "no-loop-func": 2,
+    "no-magic-numbers": [ // Reinstate this rule after the styles have been moved into seperate files
```
Comment (Ragnar-Hardarson) [inline .eslintrc]: What's this comment about?

File: README.md [hunk 6]
```diff
@@ -5,3 +5,30 @@ Web interface for the popular Politech site
...
 
 There is alot of reading material to be found in their guide [here](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md).
+
+## Yarn
+To start project
+´´´
+yarn start
+´´´
+
+Add a new package as a dependency
```
Comment (Ragnar-Hardarson) [inline README.md]: Isn't adding/removing packages basic enough that these commands are cluttering the README?

File: README.md [hunk 7]
```diff
@@ -5,3 +5,30 @@ Web interface for the popular Politech site
...
+
+Remove a new package
+´´´
+yarn remove <nameOfPackage>
+´´´
+
+## Lint
+To run ESLint
+´´´
+npm run lint
```
Comment (Ragnar-Hardarson) [inline README.md]: `yarn run lint` ?

File: src/Components/NavBar/stories/NavBar.stories.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
 /* @flow */
 import React from 'react';
-import { storiesOf, action, linkTo, addDecorator } from '@kadira/storybook';
+import { storiesOf } from '@kadira/storybook';
 import { muiTheme } from 'storybook-addon-material-ui';
-import NavBar from '../index';
+import NavBar from '../index.js';
```
Comment (Ragnar-Hardarson) [inline src/Components/NavBar/stories/NavBar.stories.js]: Which rule is this?
I'm not a fan

**Pull request merged**